### PR TITLE
#34 Finished basic Jersey support.

### PR DESCRIPTION
**Pull request by shamoh**
ISSUES:
- unnable to access ApplicationPath from CDI proxy
- Arquilian test archives contains more classes than expected

### CODE REVIEW

**Comment by shamoh** [comment]
I'm not so experienced with CDI so I've problem to get instance of ApplicationPath on the Application class that is annotated by ApplicationScoped because it is CDI proxy.

It is this problem (https://developer.jboss.org/thread/266863). I've played little bit with Extension but without success. I would appreciate your help with this.

Another issue I have is with Arquillian. I have prepared three tests to test three different deployments:
- no Application class
- Application class with NO ApplicationPath annotation
- Application class with ApplicationPath annotation

But it looks like created JavaArchive instance is not used during test.

I did not want to spend so much time on these open issue. You help is more than welcome.

File: rest-jersey/src/test/java/ws/ament/hammock/rest/jersey/RestApp.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * limitations under the License.
+ */
+
+package ws.ament.hammock.rest.jersey;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.ws.rs.core.Application;
+
+@ApplicationScoped
+public class RestApp {//extends Application { //??? Tested archive contains both Applications on classpath
```
Comment (johnament) [inline rest-jersey/src/test/java/ws/ament/hammock/rest/jersey/RestApp.java]: The arquillian extension automatically adds ws.ament.hammock.** to the archive, to help simplify usage.

File: rest-jersey/pom.xml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+            <groupId>ws.ament.hammock</groupId>
+            <artifactId>web-spi</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>javax.servlet</groupId>
+            <artifactId>javax.servlet-api</artifactId>
+        </dependency>
+        <!-- test dependencies -->
+        <dependency>
```
Comment (johnament) [inline rest-jersey/pom.xml]: Try removing this dependency and see if it makes your test better.

**Comment by johnament** [comment]
I think you're seeing https://java.net/jira/browse/JAX_RS_SPEC-534 .  I don't particularly think having a URI root makes a ton of sense in a microservices world.  I'm fine with it not working properly.

**Comment by johnament** [comment]
BTW, funny you mention that thread that you and I conversed on in the past.  I'll clarify what I meant there.  JAX-RS annotations are not inherited.  As a result, when the proxy subclasses your application instance, it gets lost.

You can still pull out the value in an extension, but that can come later.  I want to put together a rest base module.

**Comment by shamoh** [comment]
Ok, you are right, we can solve enhanced use-cases later.

I know your blog posts about micro-platform based on CDI. I like it and it inspired me to start my own and similar platform with focus on event driven approach. https://github.com/shamoh/mega But it is just hobby research project, no time to work on it regularly.

**Pull request merged**